File: QytetetJava/src/modeloqytetet/TituloPropiedad.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author fran98
 */
public class TituloPropiedad {

   private String nombre;
   private Boolean hipotecada;
   private int alquilerBase;
   private float factorRevalorizacion;
   private int hipotecaBase;
   private int precioEdificar;
   private Jugador propietario;
   private Calle calle;

   public TituloPropiedad(String nombre, int alquilerBase, float factorRevalorizacion, int hipotecaBase, int precioEdificar) {
      this.nombre = nombre;
      this.hipotecada = false;
      this.alquilerBase = alquilerBase;
      this.factorRevalorizacion = factorRevalorizacion;
      this.hipotecaBase = hipotecaBase;
      this.precioEdificar = precioEdificar;
      this.propietario = null;
      this.calle = null;
   }

   public String getNombre() {
      return nombre;
   }

   public Jugador getPropietario() {
      return propietario;
   }

   Casilla getCasilla() {
      return this.calle;
   }

   Boolean getHipotecada() {
      return hipotecada;
   }

   void setHipotecada(Boolean hipotecada) {
      this.hipotecada = hipotecada;
   }

   int getAlquilerBase() {
      return alquilerBase;
   }

   float getFactorRevalorizacion() {
      return factorRevalorizacion;
   }

   int getHipotecaBase() {
      return hipotecaBase;
   }

   int getPrecioEdificar() {
      return precioEdificar;
   }

   void setPropietario(Jugador propietario) {
      this.propietario = propietario;
   }

   void cobrarAlquiler(int coste) {
      propietario.modificarSaldo(coste);
   }

   boolean propietarioEncarcelado() {
      return propietario.isEncarcelado();
   }

   public boolean tengoPropietario() {
      return propietario != null;
   }

   void setCalle(Casilla casilla) {
      this.calle = (Calle) casilla;
   }

   @Override
   public String toString() {
      return "\n\tNombre: " + nombre
          + "\n\tHoteles: " + calle.getNumHoteles()
          + " Casas: " + calle.getNumCasas()
          + "\n\tHipotecada: " + hipotecada
          + "\n\tAlquiler base: " + alquilerBase
          + "\n\tFactor revalorizacion: " + factorRevalorizacion
          + "\n\tHipoteca base: " + hipotecaBase
          + "\n\tPrecio edificar: " + precioEdificar;
   }

}

File: QytetetRuby/lib/titulo_propiedad.rb
#encoding: UTF-8

module ModeloQytetet
  require_relative 'casilla'
  class TituloPropiedad
    attr_reader :alquiler_base, :factor_rev, :hipoteca_base,
      :hipotecada, :nombre, :precio_edificar, :casilla
    attr_writer :casilla, :hipotecada, :propietario
    def initialize (nombre, alquiler, factor, hipoteca, edificar )
      @alquiler_base = alquiler
      @factor_rev = factor
      @hipoteca_base = hipoteca
      @hipotecada = false
      @nombre = nombre
      @precio_edificar = edificar
      @propietario = nil
      @casilla = nil
    end
    def to_s
      "Titulo Propiedad: "\
        "\n\tNombre: #{@nombre}" \
        "\n\tHipotecado: #{@hipotecada}" \
        "\n\tAlquiler Base: #{@alquiler_base}" \
        "\n\tFactor Revalorización: #{@factor_rev}"\
        "\n\tHipoteca Base: #{@hipoteca_base}"\
        "\n\tPrecio Edificar: #{@precio_edificar}"\
        "\n\tpropietario: #{@propietario}"
    end
    def cobrar_alquiler(coste)
      @propietario.modificar_saldo(coste)
    end
    def propietario_encarcelado
      @propietario.encarcelado
    end
    def tengo_propietario
      @propietario != nil
    end
    def set_casilla(casilla)
      @casilla = casilla
    end
  end
end
